core/modules/wdb/QueryRunner.cc

```diff
@@ -283,6 +283,10 @@ bool QueryRunner::_fillRows(MYSQL_RES* result, int numFields, uint& rowCount, si
                 LOGS_ERROR("Message single row too large to send using protobuffer");
                 return false;
             }
+            auto totalSize = _task->addTransmitSize(tSize);
+            if (totalSize > _maxResultSize) { // &&& just forget this part on the worker ???
+                LOGS_ERROR("Result set of single query too large totalSize=" << totalSize);
+            }
```

Comment:
Don't you want to begin cancelling query execution right away once Qserv "czar" runs into this condition?
There is another problem you would need to think about - if someone accidentally does `SELECT *` on a large table w/o conditions then czar's log file will bee flooded with thousand identical messages of this kind. Perhaps you need to downgrade the logging level for this message to `TRACE`?

Reply:
I thought about this a bit. It wouldn't much faster than having the czar do it, it wound up making things a lot more complicated, and it should be a rare occurrence (i.e. I'd expect most results too large errors to come from the results of multiple chunks being too large). 

Whether the cancellation  starts at this line or on the czar when it receives this transmit, most of the result from this task will have already been sent to the czar as the worker only reads out what will be sent in the next transmit. I'll probably just get rid of this check as trying to keep configuration values consistent for both the czar and the worker is not simple.